### DISCUSSION TITLE
Cached pods code coverage

### DISCUSSION THREAD

**Discussion by kattouf** [swiftyfinch/Rugby#404] [Q&A] [answered]
Hello! We use Rugby for pre-compile and reuse all our development and 3rd party pods in CI test jobs.

Now we would like to collect code coverage, but we are faced with difficulties because the test plan "Gather coverage" applies only to the targets available in the scheme (which missed cached ones), which is why we get code coverage only for code outside of Pods (application target).

Сould you recommend something in this case?

**Comment by swiftyfinch** [comment, marked as answer]
@kattouf Hello! Is this topic relevant for now?

Anyway, I think you need some kind of test impact analysis. \
Then, to do something like this:
1. Run test impact analysis and get pods to test;
2. Exclude these pods from Rugby and run the last one;
3. Run tests;
4. And run Rugby again without these pod exclusions.

You can try to use `rugby _test impact`: \
https://github.com/swiftyfinch/Rugby/blob/main/Docs/commands-help/test/impact.md

Also, this way can't calculate code coverage between different pods. \
It's a disadvantage on the one hand, but on the other hand, it might be OK to skip code coverage for integration tests.

**Reply by kattouf** [reply]
Thank you for answer!

If I understood your proposal correctly, it might not fully meet our specific requirements. We need coverage for all development pods to analyze and provide insights/statistics to our developers. For now, we’ve decided to set up a separate CI job without Rugby caches, specifically to collect test coverage.

**Reply by swiftyfinch** [reply]
Yeah, that's much easier if it suits your needs 👍

I suggested gathering coverage for all pods, but not always in one run.
Like in incremental builds. But it's a complicated thing.
